core/coreapi/object.go

```diff
@@ -0,0 +1,190 @@
+package coreapi
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"io/ioutil"
+
+	"github.com/ipfs/go-ipfs/merkledag/utils"
+
+	coreiface "github.com/ipfs/go-ipfs/core/coreapi/interface"
+	dag "github.com/ipfs/go-ipfs/merkledag"
+	ft "github.com/ipfs/go-ipfs/unixfs"
+)
+
+type ObjectAPI CoreAPI
+
+func (api *ObjectAPI) New(ctx context.Context) (coreiface.Node, error) {
+	node := new(dag.ProtoNode)
+
+	_, err := api.node.DAG.Add(node)
+	if err != nil {
+		return nil, err
+	}
+	return node, nil
+}
+
+func (api *ObjectAPI) Put(context.Context, coreiface.Node) error {
+	return errors.New("todo") // TODO: what should this method take? Should we just redir to dag-put?f
```

Comment:
I'm not even sure if we want to keep this method, it seems to me that all of it's use cases are covered by Dag.Put

Reply:
We should still implement it -- it can call `DagAPI.Put()` underneath